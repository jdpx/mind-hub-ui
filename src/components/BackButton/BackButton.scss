@import "../../styles/colours.scss";

.back-button {
    display: flex;

    margin-bottom: 20px;

    a {
        min-height: 41px;
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 12px;
        box-shadow: 0px 4px 4px rgba(36, 59, 113, 0.5);
        border-radius: 20px;
        background-color: $background-blue;
        color: $white;

        img {
            filter: invert(95%) sepia(95%) saturate(21%) hue-rotate(15deg) brightness(105%) contrast(105%);
            margin-right: 10px;
        }
    }
}
