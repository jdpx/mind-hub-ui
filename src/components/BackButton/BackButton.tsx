import React from 'react'
import { Link } from 'react-router-dom'
import LeftArrow from '../Arrows/LeftArrow'

import './BackButton.scss'

interface Props {
    text: string
    to: string
    testid?: string
}

export default function BackButton({ text, to, testid }: Props) {
    return (
        <div className="back-button">
            <Link to={{ pathname: to }} data-testid={testid}>
                <LeftArrow height={18} />
                {text}
            </Link>
        </div>
    )
}
