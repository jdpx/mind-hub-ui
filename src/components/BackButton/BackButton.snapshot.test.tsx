import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import BackButton from './BackButton'

describe('BackButton Snapshots', () => {
    const props = {
        text: 'foo',
        to: '/testing',
    }

    it('renders back button', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <BackButton {...props} />
            </BrowserRouter>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
