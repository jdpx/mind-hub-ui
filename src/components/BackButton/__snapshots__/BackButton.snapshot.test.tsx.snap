// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Snapshots renders back button 1`] = `
<DocumentFragment>
  <div
    class="back-button"
  >
    <a
      href="/testing"
    >
      <img
        alt="Left Arrow"
        data-testid="left-arrow"
        height="18"
        src="left-arrow.svg"
      />
      foo
    </a>
  </div>
</DocumentFragment>
`;
